service/tbc/crawler.go

```diff
@@ -590,6 +598,56 @@ func (s *Server) fixupCacheBatched(ctx context.Context, b *btcutil.Block, utxos
 	return s.db.ScriptHashesByOutpoint(ctx, ops, found)
 }
 
+func (s *Server) fixupCacheChannel(ctx context.Context, b *btcutil.Block, utxos map[tbcd.Outpoint]tbcd.CacheOutput) error {
+	// prime slots
+	slots := 128
+	c := make(chan struct{}, slots)
+	defer close(c)
+	for i := 0; i < slots; i++ {
+		select {
+		case <-ctx.Done():
+			return nil
+		case c <- struct{}{}:
+		default:
+			return fmt.Errorf("shouldn't happen")
```

Comment:
Not sure why the linter didn't catch this...
```suggestion
			return errors.New("shouldn't happen")
```

Reply:
i swear i will write a benchmark one day to show this isn't worth it :-P